### ISSUE TITLE
Question: How to use it?

### ISSUE THREAD

**Issue by christian-schwaderer** [hubgit/php7-apache-saxonhe#1]
I need to get Saxon/C working on a CentOS 7 machine. So far I did not succeed.
So, I thought I'd give a Docker solution a try and thus came across your git repo.
Thanks for sharing your Dockerfile! However, I might need a few words of instruction maybe.

What I got so far:
* I set up a CentOS 7 VM with Apache2 and PHP 7. That works
* I pulled and started your docker file with `docker run hubdock/php7-apache-saxonhe`
* The output I get is

> AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
> AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
> [Sun Dec 23 16:35:14.198751 2018] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.25 (Debian) PHP/7.2.0 configured -- resuming normal operations
> [Sun Dec 23 16:35:14.198789 2018] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'

Now, how do I proceed from here? I don't have much experience with Docker so far. However, I can tell that Saxon is not yet working with this setup.

**Comment by hubgit** [comment]
This repository contains Saxon-HE compiled as a PHP extension, intended for use as a base for Docker images containing PHP applications that use Saxon, e.g. https://github.com/hubgit/xsweet-docx-html/blob/master/Dockerfile

**Comment by ond1** [comment]
Hi Christian,

Have you tried downloading Saxon/C from the website?
i.e.:  https://www.saxonica.com/download/c.xml

Also you can try the following docker file which should work for CentOS machine:

https://dev.saxonica.com/repos/archive/opensource/latest9.8/hec/Dockerfile

**Comment by christian-schwaderer** [comment]
Hi O'Neil,

thank you so much for trying to help me again!

After having commented out the line `ADD web /var/www/html` I was able to build the Dockerfile. However, when I run it, the output I get is almost the same I get for the other docker container: 
`AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
`
and then the session seems to be crashed. No reaction to any input, not even Ctr+C

Maybe I'm missing something essential, I'm quite a Docker newbie.

**Comment by ond1** [comment]
Hi,

Have you tried running php from the command line. Also using gdb debugger? Sometimes I find running a php script that way helps to find out what is going on.

kind regards,

O'Neil

**Comment by christian-schwaderer** [comment]
Hi,

many thanks again!
I got it to work. As I had assumed, the problem is just my complete lack of experience with Docker.

So, for those finding this thread, here is what I did:

```
wget -rkpN -e robots=off https://dev.saxonica.com/repos/archive/opensource/latest9.8/hec/
cd dev.saxonica.com/repos/archive/opensource/latest9.8/hec
```

With an editor (e.g. `vi`) open `Dockerfile` and comment out  `ADD web /var/www/html`

Now run
```
docker build -t mysax .
docker run -it mysax bash
```
Now inside the Docker container, run

`php -r '$saxonProc = new Saxon\SaxonProcessor(); echo $saxonProc->version();'`

it should print `Saxon/C 1.1.0 running with Saxon-HE 9.8.0.4J from Saxonica`.

Just as a suggestion: Maybe it would be good idea if you publish the container on Docker Hub with a few words of explanation?
